referenceframe/primitives_test.go

```diff
@@ -37,3 +39,34 @@ func TestInterpolateValues(t *testing.T) {
 	test.That(t, interp1, test.ShouldResemble, jpHalf)
 	test.That(t, interp2, test.ShouldResemble, jpQuarter)
 }
+
+type StateInput struct {
+	Position      spatialmath.Pose
+	Configuration []Input
+	Frame         Frame
+}
+
+var (
+	x *StateInput
+	y *StateInput
+)
+
+var (
+	inp = []Input{{1}, {2}, {3}}
+	flt = []float64{1., 2., 3.}
+)
+
+func BenchmarkCreateStruct(b *testing.B) {
```

Comment:
Do we want to keep these around?  If so they should not be living here.  Alternately, we could move `StateInput` into the primitives file